tests/server/test_compilerservice.py

```diff
@@ -598,10 +599,47 @@ async def test_server_recompile(server, client, environment, monkeypatch):
 
     # update the parameter to a new value
     await client.set_param(environment, id="param2", value="test2", source=ParameterSource.plugin, recompile=True)
-    versions = await wait_for_version(client, environment, 3)
     logger.info("wait for 3")
+    versions = await wait_for_version(client, environment, 3)
     assert versions["count"] == 3
 
+    # set a full compile schedule
+    async def schedule_soon() -> None:
+        soon: datetime.datetime = datetime.datetime.now(datetime.timezone.utc) + datetime.timedelta(seconds=2)
+        cron_soon: str = "%d %d %d * * * *" % (soon.second, soon.minute, soon.hour)
+        await client.environment_settings_set(environment, id="auto_full_compile", value=cron_soon)
+
+    async def is_compiling() -> None:
+        return (await client.is_compiling(environment)).code == 200
+
+    await schedule_soon()
+    await retry_limited(is_compiling, 3)
+    logger.info("wait for 4")
+    versions = await wait_for_version(client, environment, 4)
+    assert versions["count"] == 4
+
+    # override existing schedule
+    await schedule_soon()
+    await retry_limited(is_compiling, 3)
+    logger.info("wait for 5")
+    versions = await wait_for_version(client, environment, 5)
+    assert versions["count"] == 5
+
+    # delete schedule, verify it is cancelled
+    await schedule_soon()
+    await client.environment_setting_delete(environment, id="auto_full_compile")
+    with pytest.raises(AssertionError, match="Bounded wait failed"):
+        await retry_limited(is_compiling, 4)
+    assert (await client.list_versions(environment)).result["count"] == 5
+
+    # override with schedule in far future, check that it doesn't trigger an immediate recompile
+    recent: datetime.datetime = datetime.datetime.now(datetime.timezone.utc) - datetime.timedelta(seconds=2)
+    cron_recent: str = "%d %d * * * %d" % (recent.minute, recent.hour, recent.second)
```

Comment:
This cron expression isn't correct, is it? Or am I missing something?

Reply:
First I used the `croniter` library but it places the seconds at the end instead of the front so I switched libraries. Forgot to update this cron string.